### ISSUE TITLE
generate the init_result

### ISSUE THREAD

**Issue by 3i-cao** [woodfrog/poly-diffuse#17]
Hello author, thanks for the great work, I run the RoomFormer and see the corner and it looks different with the init_result from your repo, can you let me know the exact way you run the RoomFormer to make your init_result?

**Comment by woodfrog** [comment]
@hao-pt That's obviously a wrong claim. The diffusion model is not trained with the init_results from a specific model (e.g., RoomFormer). RoomFormer or MapTR results are only used at test time. If you read the paper or the code carefully, the input to the guidance network during training is the ground truth data. 

The reason you got poor results might simply because you run RoomFormer with a different set of hyper-parameters. 

<img width="799" height="276" alt="Image" src="https://github.com/user-attachments/assets/bfd71311-35fc-4faf-9669-230c463bb785" />

**Comment by woodfrog** [comment]
> Hello author, thanks for the great work, I run the RoomFormer and see the corner and it looks different with the init_result from your repo, can you let me know the exact way you run the RoomFormer to make your init_result?

Their code was not released when I worked on this paper, so I used my own reproduction, which produced similar (or even better?) quantitative results than those in RoomFormer's paper. The code and the checkpoint are [here](https://github.com/woodfrog/poly-diffuse/tree/main/roomformer_adapted). 


For the normal prep., you might check [here](https://github.com/woodfrog/heat/tree/master/s3d_preprocess). The original code was from MonteFloor.

**Comment by hao-pt** [comment]
Yes, I'm able to reproduce their results. Something is not right since I test PolyDiffuse with those outputs from official RoomFormer's checkpoint. I wonder if I need to change some parameter in `scripts/sample.sh`. Here is the script

```
INIT_DIR=/home/htp26/RoomFormerTest/s3d_density_preds/roomformer_nosem_test_preds/checkpoints/npy
OUTPUT_DIR=outputs/s3d_polydiffuse_step10_roomformer_nosem

python generate.py --seeds=0-1 --batch=1 --port=12347 \
--data_dir=./data/processed_s3d --init_dir=${INIT_DIR} \
--workers 8 \
--steps=10 --sigma_max=5.0 --sigma_min=0.01 --S_churn 0.0 --rho 7.0 \
--second_order=False --compute_likelihood=False \
--proposal_type=roomformer \
--guide_ckpt="training-runs/s3d_pretrained_ckpts/guide/polydiffuse-s3d-guide/network-snapshot.pth" \
--network="./training-runs/s3d_pretrained_ckpts/denoise/polydiffuse-s3d-denoise/network-snapshot.pth" \
--outdir=${OUTPUT_DIR} \
--viz_results=True
```

**Comment by woodfrog** [comment]
My RoomFormer reproduction gets slightly better quantitative results than their paper. The metric implementations should be the same because I know RoomFormer uses the same eval code as my previous project HEAT, and both are based on an earlier work MonteFloor. 

It seems that the `output scores before refinement` are much lower than the results reported in either RoomFormer paper or my paper. The qualitative results in your first screenshot also suggest the results are not right. I don't know what happened when you connecting the two codebases...  If the `init_result` is produced by running the official RoomFormer repo properly, at least those two examples should be much better.

**Comment by hao-pt** [comment]
I noticed the hardcoded logic at https://github.com/woodfrog/poly-diffuse/blob/21ef8a15637795f3555d132cc90146bc488a98f4/src/datasets/polygon_datasets.py#L242 where you assume the prediction is a closed-loop polygon and then remove the last vertex.

Here is the code to fix that:
```
if np.all(poly_item[0] == poly_item[-1]):
                processed_polygon = self.preprocess_polygon(poly_item[:-1, :].astype(np.float32))  
else:
                processed_polygon = self.preprocess_polygon(poly_item.astype(np.float32))  
```

However, there is still a mismatch of output scores between your codebase and RoomFormer. Here, I test proposal from RoomFormer's official checkpoints.

<img width="1213" height="193" alt="Image" src="https://github.com/user-attachments/assets/34c02343-4b28-4674-b3c7-f21d95fe38e8" />
